in-progress/8754-slashing-and staking.md

```diff
@@ -0,0 +1,357 @@
+
+|                      |                                   |
+| -------------------- | --------------------------------- |
+| Issue                | [title](github.com/link/to/issue) |
+| Owners               | @aminsammara                              |
+| Approvers            | @LHerskind @just-mitch @joeandrews @Maddiaa0 |
+| Target Approval Date | 2024-11-08                        |
+
+
+# Executive Summary
+
+This design introduces a Staking Mechanism and Proof Of Governance (PoG) for slashing. 
+
+For Staking, we introduce an immutable `Deposit` contract to make stake migrations more seamless.  
+
+For Slashing, the rationale behind using PoG is the use of offchain evidence for voting allows nice properties including:
+
+1) More precision in slashing. Honest node operators should not be slashed but we need offchain evidence to do this. 
+2) Provers can appeal to the validator set to unslash them in the case they could not submit their proof on time (i.e. Ethereum congestion, or an attack on proving marketplace nodes)
+
+So we rely on validators to vote to slash dishoneset actors. 
+
+# Staking
+
+Validators stake with a `Deposit` contract in order to join the validator set. They do NOT stake with the Rollup contract directly. This is meant to simplify moving stake during/after a governance upgrade. 
+
+### Deposit Contract
+
+The Deposit contract is an immutable contract living on L1, that is owned by the Governance Contract (i.e. Apella). It implements the following simplified interface:
+
+```solidity
+
+interface IDeposit {
+  function deposit(address validatorAddress, address withdrawalAddress, uint256 amountToDeposit, bool followRegistryBool, address rollupAddress) external returns(bool);
+  function activateValidator(address rollupAddress, address validatorAddress, uint256 amount);
+  function widthdraw(address rollupAddress, address validatorAddress) external returns(bool);
+  function unstakeValidators(addresss rollupAddress, address validatorAddress) external returns(bool);
+  function slashValidator(address rollupAddress, address validatorAddress) external returns(bool);
+  function getRollup() internal view returns(address);
+}
+```
+
+### Deposit Function
+
```

Comment:
Does min deposit amount imply weighted stake? or will there be some max effective balance too